File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPPlayerTime.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPPlayerTime implements PlayerStateProperty<Long> {
    @Override
    public String getConfigName() {
        return "player_time";
    }

    @Override
    public Long getDefaultSerializedProperty() {
        return null; // This value shouldn't be saved.
    }

    @Override
    public Long getSerializedPropertyFromPlayer(Player player) {
        return null; // This value shouldn't be saved.
    }

    @Override
    public void restoreProperty(Player player, Long value) {
        player.resetPlayerTime(); // Just reset.
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPPlayerTime.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPPlayerTime extends PlayerStateProperty[Long]:
  override def getConfigName: String = "player_time"

  override def getDefaultSerializedProperty: Long =
    null // This value shouldn't be saved.

  override def getSerializedPropertyFromPlayer(player: Player): Long =
    null // This value shouldn't be saved.

  override def restoreProperty(player: Player, value: Long): Unit =
    player.resetPlayerTime() // Just reset.